server/src/graphql/utils.ts

```diff
@@ -47,3 +50,80 @@ export const formatGithubPem = (pem: string) => {
 
   return formattedPem;
 };
+
+export const refreshAuthToken = async (userId: string) => {
+  const user = await prisma.user.findUnique({
+    where: {
+      id: userId,
+    },
+    select: {
+      refreshToken: true,
+      refreshTokenExpiresIn: true,
+      githubAccessToken: true,
+    },
+  });
+
+  let data;
+
+  const octo = new Octokit({});
+
+  try {
+    const res = await octo.request(`POST /login/oauth/access_token`, {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      body: JSON.stringify({
+        refresh_token: user.refreshToken,
+        grant_type: 'refresh_token',
+        client_id: config.githubAppClientId,
+        client_secret: config.githubAppClientSecret,
+      }),
+    });
+
+    data = res;
+    await prisma.user.update({
+      where: {
+        id: userId,
+      },
+      data: {
+        refreshToken: data.refresh_token,
+        refreshTokenExpiresIn: data.refresh_token_expires_in.toString(),
+        githubAccessToken: data.access_token,
+      },
+    });
+
+    return data;
+  } catch (error) {
+    console.error(error);
+    throw new Error('Request failed');
+  }
+};
+
+export const octoRequestWithUserToken = async (
+  requestData: string,
```

Comment:
can we get better typings? octokit.request is strongly typed, would be great if we could keep it somehow